unlock-app/src/components/interface/checkout/alpha/Checkout/Metadata.tsx

```diff
@@ -142,10 +143,24 @@ export function Metadata({ checkoutService, injectedProvider }: Props) {
         })
       )
 
-      await storage.submitMetadata(users, lock!.network)
-
       const recipients = users.map((item) => item.userAddress)
 
+      const generatedData = paywallConfig.dataURL
+        ? await fetchData(paywallConfig.dataURL, {
+            recipients,
```

Comment:
And let's make one call per recipient too, because the hook is per recipient

Reply:
We will need to multiple requests then for multiple recipients though.

Reply:
> We will need to make multiple requests then for multiple recipients though.
Yes that's the correct approach because that's how the hook will work